overlord/ifacestate/helpers.go

```diff
@@ -144,15 +145,39 @@ func (m *InterfaceManager) addSnaps(snaps []*snap.Info) error {
 	return nil
 }
 
-func profilesNeedRegenerationImpl() bool {
-	mismatch, err := interfaces.SystemKeyMismatch()
+func (m *InterfaceManager) profilesNeedRegeneration() bool {
+	return profilesNeedRegenerationImpl(m)
+}
+
+var profilesNeedRegenerationImpl = func(m *InterfaceManager) bool {
+	// m.useAppArmorPrompting() is supported&&enabled, rather than simply
+	// enabled, but it is okay to use it in place of enabled, since if
+	// prompting is not supported, the system key values related to
+	// prompting will both be false anyway, and if it is supported, then
+	// supported&&enabled is equal to enabled.
+	extraData := interfaces.SystemKeyExtraData{
+		PromptingFlagEnabled: m.useAppArmorPrompting(),
```

Comment:
but this is a combination of supported and enabled?  don't we have problem in that?

Reply:
The comment above `extraData` explains why it's fine in this case. Also, the doc comment for the `SystemKeyExtraData` type discusses this as well.

Reply:
https://github.com/snapcore/snapd/pull/13822/files/748125f439049c3d0447544cc644dbf35f338ca8#diff-6dee0c4fbd0fa6cd0f7240e873289ede17c3c84348ae2768dc8d74227de90916R178-R186

Reply:
Perhaps adjusting the name of the field from `PromptingFlagEnabled` to something which expresses that expresses that this can be either `enabled` or `supported && enabled` would be clearer, but I'm not sure what to call it. I think simply `Prompting` might be too vague?

Reply:
is not simply a naming problem,  if the input given in snapd and the one in snap run disagree then snap run will enter wait mode when it shouldn't

Reply:
The supported value from the previous system key is re-used if the kernel features and parser mtime are unchanged, else new profiles need to be regenerated anyway. And since the system key stores `supported` and `supported&&enabled`, and the value we pass in as `PromptingFlagEnabled` will be ANDed with `supported`, then whether `PromptingFlagEnabled` is set as `enabled` or `supported&&enabled` results in identical system keys. The values are set to:

If `PromptingFlagEnabled` is set to `(enabled)` alone:
- `AppArmorPromptingSupported`: `supported`
- `AppArmorPromptingSupportedAndEnabled`: `supported && (enabled)`

If `PromptingFlagEnabled` is set to `(supported && enabled)`:
- `AppArmorPromptingSupported`: `supported`
- `AppArmorPromptingSupportedAndEnabled`: `supported && (supported && enabled)`

So the result is identical. The opportunity for snap run to disagree with snapd is if it checks `features.AppArmorPrompting.IsEnabled()` and gets a different value from what snapd internally has set, but I believe that's the best we can do for checking an exported feature flag. Internally, the `features.AppArmorPrompting` value is only checked once, and used consistently internally for all system key-related functions (hence why we pass in a struct with the value), and snapd is restarted whenever the value changes.